src/components/CollectionsPage/CollectionsPage.tsx

```diff
@@ -31,13 +34,15 @@ import CollectionRow from './CollectionRow'
 import { Props, TABS } from './CollectionsPage.types'
 import './CollectionsPage.css'
 
+let timeout: NodeJS.Timeout
```

Comment:
As the `CollectionsPage` component lives in the web environment, we can ask for the return type of the setTimeout instead of typing it as the `NodeJS.Timeout` one:
```suggestion
let timeout: ReturnType<typeof setTimeout>
```

Reply:
Would you mind defining the `timeout` variable as being part of the `CollectionsPage` component instance so it's not globally defined?